### DISCUSSION TITLE
Tuning replication factor for sample store

### DISCUSSION THREAD

**Discussion by zkl94** [strimzi/strimzi-kafka-operator#7004] [Q&A] [answered]
I have been using strimzi for some time and one thing that is bugging me is this Grafana alert that's been popping up:

<img width="213" alt="image" src="https://user-images.githubusercontent.com/7790845/176569625-9b46d0d6-870f-4575-bdd5-203093549c2f.png">

After some investigations, I found out these alerts come from `strimzi.cruisecontrol.modeltrainingsamples` and `strimzi.cruisecontrol.partitionmetricsamples`, which only have 2 replicas when my `default.replication.factor` is set to `3`.

I also found [this issue](https://github.com/strimzi/strimzi-kafka-operator/issues/4268) that is still under discussion and Kate [proposed](https://github.com/strimzi/strimzi-kafka-operator/issues/4268#issuecomment-1098281619) that the replication factor can be tuned by changing `sample.store.topic.replication.factor: 1` under `CruiseControl.config`.

However, after changing the parameter under `CruiseControl.config`, I am getting the following warnings from the cruise-control pod (which basically says this parameter is incorrect?):

```
2022-06-30 00:30:23 WARN  ConsumerConfig:380 - The configuration 'sample.store.topic.replication.factor' was supplied but isn't a known config.
2022-06-30 00:30:23 WARN  ProducerConfig:380 - The configuration 'sample.store.topic.replication.factor' was supplied but isn't a known config.
2022-06-30 00:30:24 WARN  AdminClientConfig:380 - The configuration 'sample.store.topic.replication.factor' was supplied but isn't a known config.
```

the two Kafka topics remain unchanged, using the default upstream config:

```
kubectl get kafkatopics.kafka.strimzi.io
strimzi.cruisecontrol.modeltrainingsamples                                                         xxxx-ns   32           2                    True
strimzi.cruisecontrol.partitionmetricsamples                                                       xxxx-ns   32           2                    True
```

I also tried to tune this parameter under Kafka.spec.config and it also doesn't work.

The question is: how should I tune the replication factor for these two topics properly?

**Comment by scholzj** [comment, marked as answer]
If the topics exist, I doubt changing any settings will change anything. I think you will need to reconfigure them manually.

**Reply by zkl94** [reply]
I can see the parameters changes after deleting the already-created topics and letting them be recreated automatically
